walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,92 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
 
+/** Find available meeting time slots */
 public final class FindMeetingQuery {
+  /** Get events of requested attendees */
+  private List<Event> getRelevantEvents(Collection<Event> events, MeetingRequest request){
+    List<Event> relevantEvents = new ArrayList<>();
+    Iterator<Event> iter = events.iterator();
+    Event currEvent;
+    Collection<String> requestedAttendee = request.getAttendees();
+    Set<String> meetingAttendee = new HashSet<>();
+    while(iter.hasNext()){
+      currEvent = iter.next();
+      meetingAttendee.addAll(currEvent.getAttendees());
+      meetingAttendee.retainAll(requestedAttendee);
+      if(meetingAttendee.size() != 0){
+        relevantEvents.add(currEvent);
+      }
+      meetingAttendee.clear();
+    }
+
+    return relevantEvents;
+  }
+
+  public static final Comparator<Event> SORT_EVENT_BY_START = new Comparator<Event>() {
+    @Override
+    public int compare(Event a, Event b) {
+      return Long.compare(a.getWhen().start(), b.getWhen().start());
+    }
+  };
+
+  /** check if time from start to end is at least length of duration */
+  private boolean checkTimeRange (int start, int end, long duration) {
+    if (((long)(end - start)) < duration) {
```

Comment:
If you assign TimeRange.START_OF_DAY to a long local variable in query(), you can also pass long to this method and avoid the cast.